x-pack/plugins/fleet/common/services/agent_status_filters.ts

```diff
@@ -0,0 +1,92 @@
+/*
```

Comment:
It's a shame to have to bring all of this logic back and use the find and replace on the kuery, an alternative approach could be to do a `_search` with the runtime fields to find the agent IDs first and then do the update action?

I can see that this would make the action non-atomic and potentilaly add some complications with 10k+ agents? Just wanted to make sure we have explored other options before duplicating the status logic.

Reply:
I agree that it's not ideal to duplicate the status logic. We don't have concrete plans on making the status field permanent, right?
The search query could work, though I found that there is a [limit of 65k](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-terms-query.html) on a terms query by default, so we would have to do multiple batches to support 100k. I'll give it a try.

Reply:
Reverted the commit of adding back filters from this pr. I started refactoring to use a search first with runtime fields, and it is a bigger change, so I would prefer to do that in a separate pr. 
At least this pr fixes the Tags filters on Agent List and the update tags with default status filters.